### DISCUSSION TITLE
GeoJSON Layer is not fully displayed (in otp-react-redux) - how to debug?

### DISCUSSION THREAD

**Discussion by amenk** [maplibre/maplibre-gl-js#2172] [Q&A] [answered]
I am trying to debug this issue which appears on our OTP-React-Redux based server

https://github.com/opentripplanner/otp-react-redux/issues/778

1. https://web.addismaptransit.com/#/
2. Enter Start Bole Airport ADD
3. Enter Destination Hilton Hotel
4. Click on a route choice

It fully should display the blue line for the bus, but it's not.

When zooming in and out it gets rendered partially, after a while of repeated zooming / panning the full line gets rendered.

![otp-rr-problem](https://user-images.githubusercontent.com/1087128/218304868-d69942a8-62de-42ba-a0d6-bed3fd8067de.gif)

I had a look at the [OTP-React-Redux code ](https://github.com/opentripplanner/otp-ui/blob/2f45f1e2369ff1d78028a0936776c182928dc578/packages/transitive-overlay/src/index.tsx#L259)and there is no such thing as a rendering priority set, especially not for the different parts of the line.

So is is this a maplibre-gl-js bug? Should I open an issue or is this maybe known? (I did not find such examples using the search)

OTP-RR [developers suggest](https://github.com/opentripplanner/otp-react-redux/issues/778#issuecomment-1419185637) it might be related to the tiles we are using (it's [Makina Maps](https://github.com/makina-maps/makina-maps)) instance which is based on OpenMapTiles.

In general I am wondering how to narrow down this problem. I tried the Spector.js chrome extension to debug the WebGL layer, but its output is a bit overwhelming :-)

**Comment by HarelM** [comment, marked as answer]
I'll be surprised if this is a MapLibre bug, but everything is possible :-)
I'm not familiar with the wrapping codebase, but I don't think you'll be able to reproduce this using "vanilla" maplibre.
So I would suspect this is due to the layer that wraps maplibre.
I'm not familiar with priorities. MapLibre has sources and layers, layers are drawn one on top of the other using the given order in the style generally.
I personally have a site which presents routes using GeoJson and there's no problem showing those (very similar use case as the above, written in Angular using @maplibre/ngx-maplibre-gl).

**Reply by amenk** [reply]
Thank you for your answer - I think I found the solution :-)
The overlay generated by OTP_ReactRedux requested `  "text-font": ["Open Sans Bold", "Arial Unicode MS Bold"], ` which is not available on the tileserver, This caused the hick-ups in the rendering. If I use a font which is present on the server, it works flawlessly.
